src/App.tsx

```diff
@@ -1,61 +1,144 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import Button from 'react-bootstrap/Button';
+import Form from 'react-bootstrap/Form';
+
+import usersFromServer from './api/users';
+import { todoList, getTodoId, getUser } from './helpers/helpers';
+import { TodoList } from './components/TodoList';
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [hasTitleError, setTitleError] = useState(false);
+  const [hasUserError, setUserError] = useState(false);
+  const [todos, setTodos] = useState(todoList);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value.replace(/[^a-zA-Zа-яА-ЯіІїЇєЄ0-9\s]/g, ''));
+    setTitleError(false);
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(Number(event.target.value));
+    setUserError(false);
+  };
+
+  const clearStates = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const addTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (title.trim() && userId) {
+      const newTodo = {
+        id: getTodoId(todos),
+        title,
+        userId,
+        completed: false,
+        user: getUser(userId),
+      };
+
+      setTodos((prevTodoList) => [...prevTodoList, newTodo]);
+      clearStates();
+    }
+
+    if (!title.trim()) {
+      setTitleError(true);
+    }
+
+    if (!userId) {
+      setUserError(true);
+    }
```

Comment:
I guess will be more logical to check if data is valid and if it's not - end function work (before code of adding todo)